components/type.go

```diff
@@ -43,3 +44,19 @@ type HealthSettable interface {
 	// SetHealthy sets the health state to healthy.
 	SetHealthy() error
 }
+
+// Data is the data type that can be checked by the component.
+type Data interface {
```

Comment:
A question: what's the purpose of Summary method? Is it used to generate message for events?

Reply:
Each check in `gpud scan` output now has two values checkmark/warning sign + one-line summary. The `Summary` method can be used to print out that one-liner (e.g., `x this check has failed`).